### DISCUSSION TITLE
OMNET++ giving [osgEarth]* CropFilter - METHOD_CROPPING not available - please compile osgEarth with GEOS

### DISCUSSION THREAD

**Discussion by pirata01** [pelicanmapping/osgearth#2306] [Support Q&A] [answered]
Hi, thanks for reading my question.

I have osgearth, geos, and gdal installed using apt-get, but still get the 'subject error' when running the default omnet++ example loacted at: [https://github.com/omnetpp/omnetpp/tree/master/samples/osg-earth](https://github.com/omnetpp/omnetpp/tree/master/samples/osg-earth)

To start, here is my setup:
1. **Ubuntu**: Ubuntu 20.04.6 LTS
2. **OMNET++**: Version: 6.0, Build id: 220413-71d8fab425

The following installed using apt-get:
1. **libgeos++-dev,  libgeos-3.8.0, libgeos-c1v5, libgeos-dev, and libgeos-doc**: 3.8.0-1build1
2. **gdal-bin**: 3.0.4+dfsg-1build3
3. **openscenegraph-plugin-osgearth, libosgearth-dev**: 2.10.2+dfsg-2build2
4. **osgearth**: 2.10.2+dfsg-2build2
5. **libosgearth-dev**: 2.10.2+dfsg-2build2
_*Missing anything?_ 

Note: have the following override in OMNET++ configuration "Run/Debug settings":
1. MESA_GL_VERSION_OVERRIDE = 3.3
2. MESA_GLSL_VERSION_OVERRIDE = 330.
This allows the scene to render, otherwise the scene objects are black.

With this error, the simulation runs, however the console log gets hammered with the 'subject' error.
I tried to follow the instruction to build the osgEarth located here, but no luck:
[https://docs.osgearth.org/en/3.2/build.html](https://docs.osgearth.org/en/3.2/build.html)
* I cannot get "Step 3 - Configure CMake" working and unsure if this command is correct:
`cmake -S osgearth -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/home/patrick/Programs/vcpkg/scripts/buildsystems/vcpkg.cmake`

Any guidance would be appreciated.

Cheers,
Patrick

**Comment by gwaldron** [comment]
Even though you apparently have GEOS installed, CMake is not picking it up. Make sure your CMake configuration for osgEarth is pointing to the GEOS C libraires.

**Reply by bllgg** [reply]
Is there any specific way to point the GEOS C libraries to the CMake configurations in osgEarth?

**Reply by gwaldron** [reply, marked as answer]
Fire up cmake-gui and plug them in there?

![image](https://github.com/gwaldron/osgearth/assets/326618/b0411a31-378c-4ea4-b80a-c5fc9bd67d70)

**Reply by mpratsaki** [reply]
> Fire up cmake-gui and plug them in there?
> 
> 
> 
> ![image](https://github.com/gwaldron/osgearth/assets/326618/b0411a31-378c-4ea4-b80a-c5fc9bd67d70)
> 
> 

How can I do that?

**Reply by pirata01** [reply]
Hi,
thanks for the recommendation. I was able to get this working based on the comment about cmake-gui.
My original question is considered satisfied and my request is considered closed.
